service/src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java

```diff
@@ -320,27 +350,14 @@ public void deleteGcpCloudContext(UUID workspaceId, AuthenticatedUserRequest use
    * @return GCP project id
    */
   public String getRequiredGcpProject(UUID workspaceId) {
-    Workspace workspace = workspaceDao.getWorkspace(workspaceId);
     GcpCloudContext gcpCloudContext =
-        workspace
-            .getGcpCloudContext()
+        workspaceDao
+            .getGcpCloudContext(workspaceId)
             .orElseThrow(
```

Comment:
nit: use `.map(GcpCloudContext::getGcpProjectId)` before `.orElseThrow` and just return the whole shebang.